### PR TITLE
Refactor member API

### PR DESCRIPTION
**Pull request by ywk253100**

### CODE REVIEW

File: src/ui/api/member.go [hunk 1]
```diff
@@ -41,47 +39,82 @@ type memberReq struct {
...
-	pma.currentUserID = pma.ValidateUser()
-	mid := pma.Ctx.Input.Param(":mid")
-	if mid == "current" {
-		pma.memberID = pma.currentUserID
-	} else if len(mid) == 0 {
-		pma.memberID = 0
-	} else if len(mid) > 0 {
-		memberID, err := strconv.Atoi(mid)
+
+	if len(pma.GetStringFromPath(":mid")) != 0 {
```
Comment (reasonerjt) [inline src/ui/api/member.go]: Sorry but I really don't recall, what was the ` mid == "current"` or `len(mid) == 0` for?
And are we sure they're no longer needed?

File: src/ui/api/member.go [hunk 2]
```diff
@@ -41,47 +39,82 @@ type memberReq struct {
...
+			if err != nil {
+				text += err.Error()
+			} else {
+				text += fmt.Sprintf("%d", mid)
+			}
+			pma.HandleBadRequest(text)
+			return
+		}
+
+		member, err := dao.GetUser(models.User{
```
Comment (reasonerjt) [inline src/ui/api/member.go]: It should be my fault, the member id seems to map to user id?  This doesn't look quite right, could you open an issue for us to follow up?

Comment (ywk253100) [inline src/ui/api/member.go]: OK.

File: src/ui/api/member.go (hunk 1 above)
Comment (ywk253100) [inline src/ui/api/member.go]: I have confirmed with kun and made a global search, they are no longer needed.

File: src/ui/api/member.go (hunk 2 above)
Comment (ywk253100) [inline src/ui/api/member.go]: There is no ID in table project_member, it uses project ID and user ID as a combined primary key. So does we need to do some modification?

Comment (reasonerjt) [inline src/ui/api/member.go]: not now, please open an issue to follow up.

**Pull request merged**